### DISCUSSION TITLE
Custom fetcher syntax in Sidekiq 7?

### DISCUSSION THREAD

**Discussion by McGeekiest** [sidekiq/sidekiq#6425] [Q&A]
_Sidekiq 7.3.2 / Rails 6.1.7.8_

Hello,

I am trying to update a gem with a custom fetch to work with Sidekiq 7. The previous syntax is no longer valid:

```ruby
Sidekiq.configure_server do |config|
  config.options[:fetch] = MyFetch
end
```

This fails with:

```
undefined method `options' for #<Sidekiq::Config>
```

Apologies if I'm just missing it, but is there documentation for the refactored fetch API that would tell me how to use a custom fetch?

Thank you.

**Comment by mperham** [comment]
There's no documentation as it's considered an internal API but you can see the changes to `Sidekiq::BasicFetch` since 6.5 to grok the change.

https://github.com/sidekiq/sidekiq/blob/2451d70080db95cb5f69effcbd74381cf3b3f727/lib/sidekiq/fetch.rb#L8

**Comment by McGeekiest** [comment]
Thanks @mperham for the pointer
